web/src/settings_components.ts

```diff
@@ -501,46 +501,44 @@ export function get_widget_for_dropdown_list_settings(
     }
 }
 
-export function set_default_code_language_widget(widget: DropdownWidget): void {
-    default_code_language_widget = widget;
-}
-
-export function set_new_stream_announcements_stream_widget(widget: DropdownWidget): void {
-    new_stream_announcements_stream_widget = widget;
-}
-
-export function set_signup_announcements_stream_widget(widget: DropdownWidget): void {
-    signup_announcements_stream_widget = widget;
-}
-
-export function set_zulip_update_announcements_stream_widget(widget: DropdownWidget): void {
-    zulip_update_announcements_stream_widget = widget;
-}
-
-export function set_create_multiuse_invite_group_widget(widget: DropdownWidget): void {
-    create_multiuse_invite_group_widget = widget;
-}
-
-export function set_can_remove_subscribers_group_widget(widget: DropdownWidget): void {
-    can_remove_subscribers_group_widget = widget;
-}
-
-export function set_can_access_all_users_group_widget(widget: DropdownWidget): void {
-    can_access_all_users_group_widget = widget;
-}
-
-export function set_can_mention_group_widget(widget: DropdownWidget): void {
-    can_mention_group_widget = widget;
+export function set_dropdown_setting_widget(property_name: string, widget: DropdownWidget): void {
+    switch (property_name) {
+        case "realm_new_stream_announcements_stream_id":
+            new_stream_announcements_stream_widget = widget;
+            break;
```

Comment:
It might be possible to further refactor this to just use a map from `property_name` => `widget | null`, to avoid all this semi-duplication.